pylint/extensions/dict_init_mutate.py

```diff
@@ -0,0 +1,118 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Check for use of dictionary mutation after initialization."""
+
+from typing import TYPE_CHECKING, List
+
+from astroid import nodes
+
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import only_required_for_messages
+from pylint.interfaces import HIGH
+from pylint.lint.pylinter import PyLinter
+
+if TYPE_CHECKING:
+    from pylint.lint.pylinter import PyLinter
+
+
+def stringify_dict_items(node: nodes.NodeNG) -> str:
+    try:
+        key = node.targets[0].slice.value
+        value = node.value.value
+    except AttributeError:
+        # More complex cases will be handled later on
+        return ""
+
+    if isinstance(value, str):
+        value = f"'{value}'"
+    return f"'{key}': {value}"
```

Comment:
```suggestion
```
We can use ``node.as_string()``. Check ``RefactoringChecker._dict_literal_suggestion`` and the result in ``use-dict-literal`` functional tests.  I think it would be better to move this function to ``pylint.util`` and reuse it, it's pretty polished already. The only thing it could do better is to take care of really long terminating value when the last element is close to 60 chars.